orchestrator.py

```diff
@@ -52,11 +74,47 @@ def __init__(self, broker_ip, port=1883):
         configs = config.read_configs()
         self.device = configs["device"]
 
-    def publish_camera_status(self, message: str) -> None:
+        self.currently_logging = False
+
+        if ON_PI:
+            # Ignore warnings about multiple scripts playing with GPIO
+            gpio.setwarnings(False)
+
+            gpio.setup(logging_button_pin, gpio.IN, gpio.PUD_DOWN)
+            gpio.add_event_detect(
+                logging_button_pin, gpio.RISING, callback=self.toggle_logging
+            )
+
+            # ADC is connected to SPI bus 0, CE pin 0
+            spi = busio.SPI(clock=board.SCK, MISO=board.MISO, MOSI=board.MOSI)
+            cs = digitalio.DigitalInOut(board.CE0)
+            mcp = MCP.MCP3004(spi, cs)
+            self.battery_adc = AnalogIn(mcp, MCP.P0)
+
+    def get_battery_voltage(self) -> float:
+        return self.battery_adc.voltage * battery_calibration_factor
```

Comment:
Given that `self.battery_adc` is only set when we're running `ON_PI`, does this code error out if we're not `ON_PI`?

Reply:
Good catch, didn't even think to try running this on my PC...

I've made it not run the battery loop at all if not on the raspberry pi.